dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBoxParser.h"
+#include "Utils.h"
+
+static constexpr wstring_view c_numberBoxOperators{ L"+-*/^"sv };
+
+// Returns list of MathTokens from expression input string. If there are any parsing errors, it returns an empty vector.
+std::vector<MathToken> NumberBoxParser::GetTokens(const wchar_t* input, winrt::INumberParser numberParser)
+{
+    auto tokens = std::vector<MathToken>();
+
+    size_t index = 0;
+    bool error = false;
+    bool expectNumber = true;
```

Comment:
>bool expectNumber = true; [](start = 4, length = 25)

The parser has a lot of these sort of state variables, I think they frequently indicate that the code could be refactored, for instance I think you could write a GetNumericToken and GetOperatorToken helper methods which could remove the need for this one.  However the nature of parsers is state machines so I will not block on this. But it would be an interesting exercise to bake the stateful logic into the function hierarchy rather than these locals

Reply:
It got better since I removed error, and also I forgot I stopped using index, so that's gone too!